File: src/building/Building.js
/**
 * Building Class
 * Represents a building in the game
 */
class Building {
    static TYPES = {
        HOUSE: {
            name: 'House',
            width: 1,
            height: 1,
            buildHeight: 40,
            topColor: '#8b4513',
            leftColor: '#654321',
            rightColor: '#7a3d0f',
            sprite: 'house' // Use house sprite
        },
        TOWER: {
            name: 'Tower',
            width: 1,
            height: 1,
            buildHeight: 60,
            topColor: '#696969',
            leftColor: '#505050',
            rightColor: '#5a5a5a',
            sprite: 'building_iso' // Use iso building sprite
        },
        WAREHOUSE: {
            name: 'Warehouse',
            width: 2,
            height: 2,
            buildHeight: 30,
            topColor: '#a0522d',
            leftColor: '#8b4513',
            rightColor: '#964b1a',
            sprite: 'treehouse' // Use treehouse sprite for warehouse
        }
    };
    
    constructor(x, y, type = Building.TYPES.HOUSE, assetLoader = null) {
        this.x = x;
        this.y = y;
        this.type = type;
        this.constructed = true;
        this.health = 100;
        this.assetLoader = assetLoader;
    }
    
    /**
     * Render building
     */
    render(renderer, camera, isometricRenderer, tileScreenPos) {
        if (!this.constructed) return;
        
        // Try to use sprite if available
        let buildingSprite = null;
        if (this.assetLoader && this.type.sprite) {
            buildingSprite = this.assetLoader.getImage(this.type.sprite);
        }
        
        if (buildingSprite) {
            // Draw building sprite centered on tile
            const spriteX = tileScreenPos.x - buildingSprite.width / 2 - camera.x;
            const spriteY = tileScreenPos.y - buildingSprite.height + 16 - camera.y; // Offset to align with ground
            renderer.drawImage(
                buildingSprite,
                spriteX,
                spriteY,
                buildingSprite.width,
                buildingSprite.height
            );
        } else {
            // Fallback to isometric cube rendering
            isometricRenderer.drawIsometricCube(
                tileScreenPos.x,
                tileScreenPos.y,
                64, // tileWidth
                32, // tileHeight
                this.type.buildHeight,
                this.type.topColor,
                this.type.leftColor,
                this.type.rightColor,
                camera
            );
        }
    }
    
    /**
     * Get building info
     */
    getInfo() {
        return {
            name: this.type.name,
            position: { x: this.x, y: this.y },
            health: this.health
        };
    }
}

File: cpp/src/building/Building.cpp
#include "building/Building.h"

Building::Building(int x, int y, BuildingType type)
    : x(x)
    , y(y)
    , type(type)
{
    initializeFromType();
}

std::string Building::getTypeName() const {
    switch (type) {
        case BuildingType::HOUSE:     return "House";
        case BuildingType::TOWER:     return "Tower";
        case BuildingType::WAREHOUSE: return "Warehouse";
        default:                      return "Unknown";
    }
}

void Building::initializeFromType() {
    switch (type) {
        case BuildingType::HOUSE:
            width = 2;
            height = 2;
            buildHeight = 40.0f;
            topColor = glm::vec4(0.8f, 0.2f, 0.2f, 1.0f);
            leftColor = glm::vec4(0.6f, 0.15f, 0.15f, 1.0f);
            rightColor = glm::vec4(0.7f, 0.17f, 0.17f, 1.0f);
            break;
            
        case BuildingType::TOWER:
            width = 1;
            height = 1;
            buildHeight = 80.0f;
            topColor = glm::vec4(0.5f, 0.5f, 0.5f, 1.0f);
            leftColor = glm::vec4(0.3f, 0.3f, 0.3f, 1.0f);
            rightColor = glm::vec4(0.4f, 0.4f, 0.4f, 1.0f);
            break;
            
        case BuildingType::WAREHOUSE:
            width = 3;
            height = 3;
            buildHeight = 30.0f;
            topColor = glm::vec4(0.6f, 0.4f, 0.2f, 1.0f);
            leftColor = glm::vec4(0.45f, 0.3f, 0.15f, 1.0f);
            rightColor = glm::vec4(0.52f, 0.35f, 0.17f, 1.0f);
            break;
    }
}
